src/transformers/models/longformer/modeling_longformer.py

```diff
@@ -1697,18 +1708,14 @@ def forward(
             embedding_output,
             attention_mask=extended_attention_mask,
             head_mask=head_mask,
+            padding_len=padding_len,
             output_attentions=output_attentions,
             output_hidden_states=output_hidden_states,
             return_dict=return_dict,
         )
         sequence_output = encoder_outputs[0]
         pooled_output = self.pooler(sequence_output) if self.pooler is not None else None
 
-        # undo padding
-        if padding_len > 0:
-            # unpad `sequence_output` because the calling function is expecting a length == input_ids.size(1)
-            sequence_output = sequence_output[:, :-padding_len]
-
```

Comment:
Move the unpadding logic to `LongformerEncoder`, so we don't need to deal with `encoder_outputs[1:]` or `encoder_outputs[2:]` here (which depends on the combination of `output_hidden_states`/`output_attentions`).